tests/src/test/scala/com/nvidia/spark/rapids/GeneratedInternalRowToCudfRowIteratorRetrySuite.scala

```diff
@@ -164,7 +164,7 @@ class GeneratedInternalRowToCudfRowIteratorRetrySuite
         ctriter, schema, TargetSize(1),
         NoopMetric, NoopMetric, NoopMetric, NoopMetric, NoopMetric)
       RmmSpark.forceSplitAndRetryOOM(RmmSpark.getCurrentThreadId)
-      assertThrows[GpuSplitAndRetryOOM] {
```

Comment:
I think the code in the next could throw either a CpuSplitAndRetryOOM or a GpuSplitAndRetryOOM. You mentioned this was in draft for testing. It may be nice to trigger both OOMs in the unit tests.

Reply:
There is no good way to do that.  Injecting an OOM is the very next allocation attempt CPU or GPU. We would have to rewrite the injection code + APIs to do that.

Reply:
I wonder if we could get around this by adding another API that allows us to disable forced OOMs for either cpu or gpu?   Then in tests we could disable one side or the other to make it possible to test specifically cpu or gpu?

Reply:
Ah ok, I missed this in spark-rapids-jni. I think it may be worth as a follow on. What do you think?

Reply:
I totally agree that we need better control over how we insert failure so we can get more test coverage. But I think that is generally true even before the CPU code was added in. If we only ever inject a failure on the first allocation we don't know what happens if the second failed instead.

Reply:
https://github.com/NVIDIA/spark-rapids-jni/issues/1609